app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,20 @@
+class MoviesController < ApplicationController
+
+  def show
+    @movie = Movie.find(params[:id])
+    @actors = @movie.actors
+    @actors_sorted_by_age = @movie.sort_actors_by_age
+    @average_age_of_actors = @movie.average_age_of_actors
+  end
+
+  def update
+    @movie = Movie.find(params[:id])
+
+    if params[:actor_id].present?
+      actor = Actor.find_by(id: params[:actor_id])
+      @movie.add_actor(actor.id)
+
+      redirect_to @movie
+    end
+  end
```

Comment:
Taking a look at your implementation for US 3, it's obvious what you are trying to do and it logically makes sense but we are jumping through a lot of hoops to get to the final destination.

Let's think about our end goal, the goal of US 3 is to link an Actor with a  Movie, and where do those links live? On the joins table `actor_movies`. And because we are CREATING a new link between them, this is going to go in the `create` action, but on what controller? Since we are not creating a new movie, it wouldn't go in the `MoviesController`, and we are not creating a new actor so it probably won't go in the `ActorsController` either. It makes sense for it to go in the `ActorMoviesController` because we are creating a link between an actor and a movie.

```ruby
class ActorMoviesController < ApplicationController
  def create
    ActorMovie.create(movie_id: params[:id], actor_id: params[:actor_id])

    redirect_to "/movies/#{params[:id]}"
  end
end
```

Doing it this way eliminates the need to create a new model method in the `Movie` model, testing that method, having to check for the presence of the `:actor_id`, and all of the other stuff you had to do to get this to work.